rtos/pulpos/pulp/rules/pulpos/targets/pulp_cv32e40p.mk

```diff
@@ -1,10 +1,14 @@
 CONFIG_NB_CLUSTER_PE ?= 8
 
-PULP_LDFLAGS      += 
-PULP_CFLAGS       +=  -D__riscv__ -DARCHI_HAS_COREV -DPLP_NO_BUILTIN
+# Define the toolchain under use with CV32E40P. Options are `PULP_TOOLCHAIN` or `COREV_TOOLCHAIN`
+TOOLCHAIN = PULP_TOOLCHAIN
```

Comment:
The users should not modify the architecture specific makefile to avoid confusion. I propose something like the following:

PULP_CC      ?= $(shell (basename $(PULP_RISCV_GCC_TOOLCHAIN)/bin/*-gcc))
PULP_LD      ?= $(shell (basename $(PULP_RISCV_GCC_TOOLCHAIN)/bin/*-gcc))
PULP_OBJDUMP ?= $(shell (basename $(PULP_RISCV_GCC_TOOLCHAIN)/bin/*-objdump))
PULP_AR      ?= $(shell (basename $(PULP_RISCV_GCC_TOOLCHAIN)/bin/*-ar))

ifeq ($(PULP_CC), riscv32-corev-elf-gcc)
__COREV_TOOLCHAIN_FLAGS__
else
__PULP_TOOLCHAIN_FLAGS__
endif